programs/uxd/src/instructions/credix_lp/rebalance_request_create_from_credix_lp_depository.rs

```diff
@@ -0,0 +1,283 @@
+
+    /// #5
+    #[account(
+        mut,
+        seeds = [
+            CREDIX_LP_DEPOSITORY_NAMESPACE,
+            depository.load()?.credix_global_market_state.key().as_ref(),
+            depository.load()?.collateral_mint.as_ref()
+        ],
+        bump = depository.load()?.bump,
+        has_one = controller @UxdError::InvalidController,
+        has_one = collateral_mint @UxdError::InvalidCollateralMint,
+        has_one = depository_shares @UxdError::InvalidDepositoryShares,
+        has_one = credix_global_market_state @UxdError::InvalidCredixGlobalMarketState,
+        has_one = credix_signing_authority @UxdError::InvalidCredixSigningAuthority,
+        has_one = credix_liquidity_collateral @UxdError::InvalidCredixLiquidityCollateral,
+        has_one = credix_shares_mint @UxdError::InvalidCredixSharesMint,
+    )]
+    pub depository: AccountLoader<'info, CredixLpDepository>,
+
+    /// #6
+    pub collateral_mint: Box<Account<'info, Mint>>,
+
+    /// #7
+    #[account(mut)]
+    pub depository_shares: Box<Account<'info, TokenAccount>>,
+
+    /// #8
+    #[account()]
+    pub credix_global_market_state: Box<Account<'info, credix_client::GlobalMarketState>>,
+
+    /// #9
+    /// CHECK: unused by us, checked by credix
+    pub credix_signing_authority: AccountInfo<'info>,
+
+    /// #10
+    #[account(mut)]
+    pub credix_liquidity_collateral: Box<Account<'info, TokenAccount>>,
+
+    /// #11
+    #[account(mut)]
+    pub credix_shares_mint: Box<Account<'info, Mint>>,
+
+    /// #12
+    #[account(
+        owner = credix_client::ID,
+        seeds = [
+            credix_global_market_state.key().as_ref(),
+            depository.key().as_ref(),
+            CREDIX_LP_EXTERNAL_PASS_NAMESPACE
+        ],
+        bump,
+        seeds::program = credix_client::ID,
+        constraint = credix_pass.user == depository.key() @UxdError::InvalidCredixPass,
+        constraint = credix_pass.disable_withdrawal_fee @UxdError::InvalidCredixPassNoFees,
+    )]
+    pub credix_pass: Account<'info, credix_client::CredixPass>,
+
+    /// #13
+    #[account(
+        mut,
+        owner = credix_client::ID,
+        seeds = [
+            credix_global_market_state.key().as_ref(),
+            &credix_global_market_state.latest_withdraw_epoch_idx.to_le_bytes(),
+            CREDIX_LP_EXTERNAL_WITHDRAW_EPOCH_NAMESPACE
+        ],
+        bump,
+        seeds::program = credix_client::ID,
+    )]
+    pub credix_withdraw_epoch: Account<'info, credix_client::WithdrawEpoch>,
+
+    /// #14
+    /// CHECK: initialized by credix program, not manipulated by us
+    #[account(mut)]
+    pub credix_withdraw_request: AccountInfo<'info>,
+
+    /// #15
+    pub system_program: Program<'info, System>,
+    /// #16
+    pub credix_program: Program<'info, credix_client::program::Credix>,
+}
+
+pub(crate) fn handler(ctx: Context<RebalanceRequestCreateFromCredixLpDepository>) -> Result<()> {
```

Comment:
What happens if we call this instruction twice within the same withdraw request window? Does it overrides? Create a second request?

Reply:
ha good one, missed this

Reply:
Integration tested over here:
- https://github.com/UXDProtocol/uxd-program/pull/244/files#diff-80dc658a8b356d2727e5b2f9ec2d33568de94bbb63ec3064fc6a6aa5fd9ef953R222

Basically:
- calling 2 create requests in a row will fail, only the first one succeeds
- calling 2 execute requests in a row will succeed, but the amount from the first execution withdrawn is substracted from the second call moved amount